cmd/kwil-admin/cmds/snapshot/create.go

```diff
@@ -243,32 +251,59 @@ func pgDump(ctx context.Context, dbName, dbUser, dbPass, dbHost, dbPort string,
 				// Write the sanitized line to the gzip writer
 				n, err := multiWriter.Write([]byte(line + "\n"))
 				if err != nil {
-					return nil, fmt.Errorf("failed to write to gzip writer: %w", err)
+					return -1, nil, fmt.Errorf("failed to write to gzip writer: %w", err)
 				}
 				totalBytes += int64(n)
 			}
 		}
 	}
 
 	if err := scanner.Err(); err != nil {
-		return nil, fmt.Errorf("failed to scan pg_dump output: %w", err)
+		return -1, nil, fmt.Errorf("failed to scan pg_dump output: %w", err)
 	}
 
 	// Close the writer when pg_dump completes to signal EOF to sed
 	if err := pgDumpCmd.Wait(); err != nil {
-		return nil, errors.New(stderr.String())
+		return -1, nil, errors.New(stderr.String())
 	}
 
+	// Append the below sql statement to the dump file to adjust the expiration times of the resolutions
+	// This is to ensure that the resolutions are correctly expired on the new network
+	n, err := multiWriter.Write([]byte("UPDATE kwild_voting.resolutions SET expiration = expiration-" + strconv.FormatInt(height, 10) + ";\n"))
+	if err != nil {
+		return -1, nil, fmt.Errorf("failed to write resolution updates to gzip writer: %w", err)
+	}
+
+	totalBytes += int64(n)
+
 	gzipWriter.Flush()
 	hash := hasher.Sum(nil)
 	genCfg.DataAppHash = hash
 
 	// Write the genesis config to a file
 	genesisFile := filepath.Join(snapshotDir, "genesis.json")
 	if err := genCfg.SaveAs(genesisFile); err != nil {
-		return nil, fmt.Errorf("failed to save genesis config: %w", err)
+		return -1, nil, fmt.Errorf("failed to save genesis config: %w", err)
 	}
 
-	return []string{fmt.Sprintf("Snapshot created at: %s, Total bytes written: %d", dumpFile, totalBytes),
+	return height, []string{fmt.Sprintf("Snapshot created at: %s, Total bytes written: %d", dumpFile, totalBytes),
 		fmt.Sprintf("Genesis config created at: %s, Genesis hash: %s", genesisFile, fmt.Sprintf("%x", hash))}, nil
 }
+
+func chainHeight(ctx context.Context, dbName, dbUser, dbPass, dbHost, dbPort string) (int64, error) {
+	addr := net.JoinHostPort(dbHost, dbPort)
+	connStr := fmt.Sprintf("postgres://%s:%s@%s/%s", dbUser, dbPass, addr, dbName)
+	conn, err := pgx.Connect(ctx, connStr)
```

Comment:
Sensible approach to updating resolution expiration heights, but ideally neither pgx nor the `meta` schema knowledge would escape those packages.  Can we use `pg.NewPool` and `meta.GetChainState` instead?

Reply:
Yeah sure.